Makefile

```diff
@@ -278,11 +278,15 @@ out/doc/api/assets/%: doc/api_assets/% out/doc/api/assets/
 out/doc/%: doc/%
 	cp -r $< $@
 
+# check if ./node is actually set, else use user pre-installed binary
+gen-json = tools/doc/generate.js --format=json $< > $@
 out/doc/api/%.json: doc/api/%.md
-	$(NODE) tools/doc/generate.js --format=json $< > $@
+	[ $(NODE) ] && $(NODE) $(gen-json) || node $(gen-json)
```

Comment:
This won’t really work like you want it to, `[ $(foobar) ]` only checks that `foobar` is nonempty, not that it corresponds . It should probably be something like `[ -x "$(NODE)" ]` to make sure that the file is there + executable.

Or you just leave the check out, `||` takes care of a failure in the first version anyway…